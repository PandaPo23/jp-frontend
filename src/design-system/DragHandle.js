import system from '../utils/System';
import IconButton from './IconButton';

const DragHandle = system('DragHandle', {
  extend: IconButton,
  position: 'absolute',
  name: 'drag-vertical',
  py: 3,
  px: 0,
  top: '50%',
  transform: 'translate(0,-50%)',
  bg: 'secondary',
  right: 0,
  cursor: 'col-resize',
});

export default DragHandle;
