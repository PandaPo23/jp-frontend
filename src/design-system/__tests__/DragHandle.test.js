import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DragHandle from '../DragHandle';

describe('DragHandle component', () => {
  it('should render as DragHandle', () => {
    const wrapper = mount(<DragHandle />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('simulates click events', () => {
    const onClick = jest.fn();
    const localProps = {
      onClick: onClick,
    };
    const wrapper = mount(<DragHandle {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper
      .find(DragHandle)
      .first()
      .simulate('click');
    expect(onClick).toBeCalled();
  });

  // it('should render disabled DragHandle', () => {
  //   const onClick = jest.fn();
  //   const wrapper = mount(<DragHandle onClick={onClick} disabled />);
  //   expect(toJson(wrapper)).toMatchSnapshot();
  //   wrapper
  //     .find(DragHandle)
  //     .first()
  //     .simulate('click');

  //   expect(onClick).not.toBeCalled();
  // });
});
