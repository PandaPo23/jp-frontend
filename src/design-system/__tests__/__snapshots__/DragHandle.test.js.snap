// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragHandle component should render as DragHandle 1`] = `
.emotion-4 {
  color: misc.muted;
  background-color: secondary;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 0;
  padding-right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  position: absolute;
  right: 0;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  cursor: col-resize;
}

.emotion-0 {
  min-width: 24px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 24px;
}

<DragHandle
  bg="secondary"
  cursor="col-resize"
  name="drag-vertical"
  position="absolute"
  px={0}
  py={3}
  right={0}
  top="50%"
  transform="translate(0,-50%)"
>
  <IconButton
    bg="secondary"
    className="emotion-5"
    cursor="col-resize"
    name="drag-vertical"
    position="absolute"
    px={0}
    py={3}
    right={0}
    top="50%"
    transform="translate(0,-50%)"
  >
    <Flex
      alignItems="center"
      as="button"
      bg="secondary"
      border="none"
      className="emotion-5"
      color="misc.muted"
      cursor="col-resize"
      disabled={false}
      display="flex"
      flexDirection="column"
      flexWrap="nowrap"
      hoverable={true}
      justifyContent="center"
      nobr={true}
      position="absolute"
      px={0}
      py={3}
      right={0}
      top="50%"
      transform="translate(0,-50%)"
    >
      <button
        className="emotion-4"
        cursor="col-resize"
        disabled={false}
        transform="translate(0,-50%)"
      >
        <Icon
          name="drag-vertical"
          order={1}
          size={24}
        >
          <Base
            className="emotion-0"
            name="drag-vertical"
            order={1}
            size={24}
          >
            <Flex
              centered={true}
              className="emotion-0"
              display="flex"
              order={1}
            >
              <div
                className="emotion-1"
              >
                <svg
                  className="emotion-0"
                  name="drag-vertical"
                  order={1}
                  size={24}
                />
              </div>
            </Flex>
          </Base>
        </Icon>
      </button>
    </Flex>
  </IconButton>
</DragHandle>
`;

exports[`DragHandle component simulates click events 1`] = `
.emotion-4 {
  color: misc.muted;
  background-color: secondary;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 0;
  padding-right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  position: absolute;
  right: 0;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  cursor: col-resize;
}

.emotion-0 {
  min-width: 24px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 24px;
}

<DragHandle
  bg="secondary"
  cursor="col-resize"
  name="drag-vertical"
  onClick={[MockFunction]}
  position="absolute"
  px={0}
  py={3}
  right={0}
  top="50%"
  transform="translate(0,-50%)"
>
  <IconButton
    bg="secondary"
    className="emotion-5"
    cursor="col-resize"
    name="drag-vertical"
    onClick={[MockFunction]}
    position="absolute"
    px={0}
    py={3}
    right={0}
    top="50%"
    transform="translate(0,-50%)"
  >
    <Flex
      alignItems="center"
      as="button"
      bg="secondary"
      border="none"
      className="emotion-5"
      color="misc.muted"
      cursor="col-resize"
      disabled={false}
      display="flex"
      flexDirection="column"
      flexWrap="nowrap"
      hoverable={true}
      justifyContent="center"
      nobr={true}
      onClick={[MockFunction]}
      position="absolute"
      px={0}
      py={3}
      right={0}
      top="50%"
      transform="translate(0,-50%)"
    >
      <button
        className="emotion-4"
        cursor="col-resize"
        disabled={false}
        onClick={[MockFunction]}
        transform="translate(0,-50%)"
      >
        <Icon
          name="drag-vertical"
          order={1}
          size={24}
        >
          <Base
            className="emotion-0"
            name="drag-vertical"
            order={1}
            size={24}
          >
            <Flex
              centered={true}
              className="emotion-0"
              display="flex"
              order={1}
            >
              <div
                className="emotion-1"
              >
                <svg
                  className="emotion-0"
                  name="drag-vertical"
                  order={1}
                  size={24}
                />
              </div>
            </Flex>
          </Base>
        </Icon>
      </button>
    </Flex>
  </IconButton>
</DragHandle>
`;
